packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionHistory.tsx

```diff
@@ -154,6 +155,12 @@ export const TransactionHistory = ({
 
   return (
     <div className="flex flex-col justify-around gap-6">
+      {isNetwork(l2.network.id).isOrbitChain && (
+        <CustomMessageWarning>
+          Fetching transaction history data and transaction execution might be
```

Comment:
It's slower because Orbit chains correspond their block range to a specific L1 block, instead of an Arbitrum block.
With L1 <> L2 interactions it's not a problem, if Arbitrum block is in the range of 100 and 200, and it corresponds to L1 block 10, we can fetch it simply.

With Orbit, they wouldn't correspond to Arbitrum, which means we have to look up L2 block for the specific Orbit range, which is extra computation.

If you're interested to know more check this PR out: https://github.com/OffchainLabs/arbitrum-sdk/pull/317